### ISSUE TITLE
Create complex types defined by default

### ISSUE THREAD

**Issue by ben-manes** [joelittlejohn/jsonschema2pojo#689]
This builds onto the discussion from #664. In this case I have a complex type to default, e.g.

```json
"active": {
  "items": {
    "$ref": "#/definitions/edge"
  },
  "type": "array",
  "uniqueItems": true,
  "default": [
    { "entityId": "11111111-0000-0000-0000-000000000000", "displayName": "Entity" }
  ]
}
```

Ideally the generated type would have,
```java
private Set<Edge> active = new LinkedHashSet<Edge>(Arrays.asList(new Edge(UUID.create("11111111-0000-0000-0000-000000000000"), "Entity");
```

but it generates,
```java
private Set<Edge> active = new LinkedHashSet<Edge>(Arrays.asList(null));
```

Is it possible to generate complex types?

**Comment by joelittlejohn** [comment]
I think to support arbitrarily complex values for the default, we'd have to start using something like the Jackson ObjectMapper (or Gson, depending over what annotator is being used I guess). We could parse the value into a static field then reference this field as the default for instance fields.

**Comment by joelittlejohn** [comment]
I mean using the ObjectMapper *in the generated source*.

**Comment by ben-manes** [comment]
Yes, it might not be worthwhile especially if deserialization is required when creating a new instance. But perhaps then the observation here is that `null` is specified as an element in the collection. If js2p cannot create the value, I don't think null is a friendly alternative.

**Comment by joelittlejohn** [comment]
@ben-manes Yes, I guess we should remove the null here and simply ignore defaults that we can't map (for the time being).

**Comment by giordanolucas** [comment]
Is there any way to avoid initializing the collections with its default values?
![image](https://user-images.githubusercontent.com/27356984/45896886-df681e00-bdab-11e8-8100-82d53f4a3fac.png)
Right now I'm getting this invalid code that I would like to avoid

**Comment by joelittlejohn** [comment]
There is no way to ignore the defaults I'm afraid, although I'd welcome a patch.
